tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -325,6 +326,38 @@ private void FeaturesProductConfigChanged(object sender, ProductConfigChangedEve
             e.Acknowledge(features.Name);
         }
 
+        private void AsmDataProductConfigRemoved(object sender, ProductConfigChangedEventArgs e)
+        {
+            if (!_enabled)
```

Comment:
I've always wondered, but I know we have this condition for all other products changes, if appsec is disabled via RCM, some updates are sent in the meantime, then appsec is re enabled, will we get these updates again after activation? 🤯

Reply:
it's a good point, we should store the data, but not pass it to the WAF fixed in 9829c3701